src/h5web/providers/mock/data.ts

```diff
@@ -0,0 +1,194 @@
+
+export function makeSimpleShape(dims: HDF5Dims): HDF5SimpleShape {
+  return { class: HDF5ShapeClass.Simple, dims };
+}
+
+/* ---------------------- */
+/* ----- ATTRIBUTES ----- */
+
+export function makeStrAttr(name: string, value: HDF5Value): HDF5Attribute {
+  return { name, value, type: stringType, shape: scalarShape };
+}
+
+/* -------------------- */
+/* ----- ENTITIES ----- */
+
+export function makeDataset(
+  id: HDF5Id,
+  type: HDF5Type,
+  shape: HDF5Shape,
+  attributes?: HDF5Attribute[]
+): HDF5Dataset {
+  return { id, collection: HDF5Collection.Datasets, type, shape, attributes };
+}
+
+export function makeSimpleDataset(
+  id: HDF5Id,
+  type: HDF5Type,
+  dims: HDF5Dims,
+  attributes?: HDF5Attribute[]
+): HDF5Dataset {
+  return makeDataset(id, type, makeSimpleShape(dims), attributes);
+}
+
+export function makeGroup(
+  id: HDF5Id,
+  attributes: HDF5Attribute | HDF5Attribute[] = [],
+  links: HDF5Link | HDF5Link[] = []
+): HDF5Group {
+  const attributesArr = Array.isArray(attributes) ? attributes : [attributes];
+  const linksArr = Array.isArray(links) ? links : [links];
+
+  return {
+    id,
+    collection: HDF5Collection.Groups,
+    attributes: attributesArr,
+    links: linksArr,
+  };
+}
+
+export function makeDatatype(id: HDF5Id, type: HDF5Type): HDF5Datatype {
+  return { id, collection: HDF5Collection.Datatypes, type };
+}
+
+export function withAttributes<T extends HDF5Dataset | HDF5Group>(
+  datasetOrGroup: T,
+  attributes: HDF5Attribute | HDF5Attribute[]
+): T {
+  return {
+    ...datasetOrGroup,
+    attributes: [
+      ...(datasetOrGroup.attributes || []),
+      ...(Array.isArray(attributes) ? attributes : [attributes]),
+    ],
+  };
+}
+
+/* ----------------- */
+/* ----- LINKS ----- */
+
+export function makeDatasetHardLink(title: string, id: string): HDF5HardLink {
+  return {
+    class: HDF5LinkClass.Hard,
+    title,
+    collection: HDF5Collection.Datasets,
+    id,
+  };
+}
+
+export function makeGroupHardLink(title: string, id: string): HDF5HardLink {
+  return {
+    class: HDF5LinkClass.Hard,
+    title,
+    collection: HDF5Collection.Groups,
+    id,
+  };
+}
+
+/* ----------------- */
+/* ----- NEXUS ----- */
+
+export function makeNxDataGroup(
+  id: string,
+  opts: {
+    signal: string;
+    axes?: string | string[];
+    ids?: Record<string, HDF5Id>;
+  }
+): HDF5Group {
+  const { signal, axes, ids = {} } = opts;
+
+  return makeGroup(
+    id,
+    [
+      makeStrAttr('NX_class', 'NXdata'),
+      makeStrAttr('signal', signal),
+      ...(axes ? [makeStrAttr('axes', axes)] : []),
+    ],
+    Object.entries(ids).map(([...args]) => makeDatasetHardLink(...args))
+  );
+}
+
+/* -------------------- */
+/* ----- METADATA ----- */
+
+export function makeMetadata(
+  opts: {
+    root?: HDF5Id;
+    datasets?: HDF5Dataset | HDF5Dataset[];
+    groups?: HDF5Group | HDF5Group[];
+    datatypes?: HDF5Datatype | HDF5Datatype[];
```

Comment:
Same, an `opts` object allows me to pass only some of these parameters in any order.